src/panels/ConfigPanel/ConfigPanel.tsx

```diff
@@ -57,14 +70,49 @@ enum InlineErrors {
   SUBMIT_TO_JUJU = "submit-to-juju",
 }
 
+enum ConfigConfirmType {
+  GRANT = "grant",
+}
+
+type ConfirmTypes = DefaultConfirmTypes | ConfigConfirmType;
+
 type ConfigQueryParams = {
   panel: string | null;
   charm: string | null;
   entity: string | null;
   modelUUID: string | null;
 };
 
+const getRequiredGrants = (
+  appName: string,
+  config: Config,
+  secrets?: ListSecretResult[] | null,
+) => {
+  const secretURIs = Object.values(config).reduce<string[]>((uris, entry) => {
+    const value = entry.newValue ?? entry.value;
+    if (
+      value &&
+      typeof value === "string" &&
+      (entry.type === "secret" ||
+        (entry.type === "string" && value.startsWith("secret:")))
+    ) {
+      uris.push(value);
+    }
+    return uris;
+  }, []);
+  return secrets
+    ? secretURIs?.filter((secretURI) => {
+        const secret = secrets.find(({ uri }) => uri === secretURI);
+        const access = secret?.access?.find(
```

Comment:
I've observed that a secret that has an owner app doesn't have the `access` key (see attached screenshot).

![Screenshot from 2024-02-19 21-15-39](https://github.com/canonical/juju-dashboard/assets/108150922/97491fa4-1410-4b6e-befe-13470c21906d)

Because of this, we will add this secret to the `getRequiredGrants(...)` list. Also, we might add the same secret multiple times, as the same secret might come from different config options (see attached screenshot).

![Screenshot from 2024-02-19 21-18-51](https://github.com/canonical/juju-dashboard/assets/108150922/d27a8c2a-c80d-4a8f-964f-69822a31d784)

Reply:
> I've observed that a secret that has an owner app doesn't have the `access` key (see attached screenshot).
> Because of this, we will add this secret to the `getRequiredGrants(...)` list, even if it has been granted permission to beforehand.

We shouldn't grant to app-owned secrets (this will be fixed in the API) so I've filtered them out when checking for required grants. I'll do another PR to remove them from the picker.


> Also, we might add the same secret multiple times to this list, as the same secret might come from different config options (see attached screenshot). I think it might make sense to remove duplicates here.

Thanks, I've deduped the apps when granting.